app/policies/grouping_policy.rb

```diff
@@ -75,7 +75,7 @@ def download_starter_file?
     return false if record.assignment.is_hidden?
     return false if !record.assignment.starter_files_after_due && record.past_collection_date?
     return true unless record.assignment.is_timed?
-
+    record.assignment ?  !check?(:see_hidden?, record.assignment) : !check?(:see_hidden?, record.grade_entry_form)
```

Comment:
groupings are only ever associated to assignments, not to grade entry forms so you can simplify this to:

```ruby
!check?(:see_hidden?, record.assignment)
```

Also, I'm assuming that this is meant to replace the check on line 75 (`return false if record.assignment.is_hidden?`) so you should replace that line instead of adding it to the bottom. Something like:

```ruby
return false unless check?(:see_hidden?, record.assignment)
```